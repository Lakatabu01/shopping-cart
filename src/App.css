.App {
  min-height: 100vh;
  position: relative;
  margin: 0;
  background-color: rgb(233, 231, 231);
}

