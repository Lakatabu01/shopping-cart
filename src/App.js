import "./App.css";
import React, { useState, useEffect } from "react";
import { Header } from "./components/header";
import { BrowserRouter, Routes, Route } from "react-router-dom";
import { Store } from "./components/store";
import { Home } from "./components/home";
import { About } from "./components/about";
import { Footer } from "./components/footer";

function App() {
  const [itemsPurchased, setItemsPurchased] = useState(0);
  const [productsSelected, setProductsSelected] = useState([]);

  useEffect(() => {
    console.log(productsSelected);
  }, [productsSelected]);

  //Get information of the products added to cart and
  //put it in a state
  const handleClick = (e) => {
    setItemsPurchased(itemsPurchased + 1);
    let productInfo = {
      image: e.target.parentNode.previousElementSibling.firstChild.src,
      productTitle:
        e.target.parentNode.previousElementSibling.children[1].textContent,
      price: parseInt(
        e.target.parentNode.previousElementSibling.children[2].textContent
      ),
    };
    setProductsSelected((prevProducts) => [...prevProducts, productInfo]);
  };

  return (
    <BrowserRouter>
      <div data-testid="test-app" className="App">
        <Header quantity={itemsPurchased} />
        <Routes>
          <Route path="/" element={<Home />} />
          <Route
            path="/store"
            element={<Store count={handleClick} items={itemsPurchased} />}
          />
          <Route path="/about" element={<About />} />
        </Routes>
        <Footer />
      </div>
    </BrowserRouter>
  );
}

export default App;
