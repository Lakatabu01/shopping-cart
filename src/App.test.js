import { render, screen } from "@testing-library/react";
import App from "./App";

test("everything rendered on the app", () => {
  const { app } = render(<App />);
  expect(app).toMatchSnapshot();
});
