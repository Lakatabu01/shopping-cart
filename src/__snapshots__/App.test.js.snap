// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`everything rendered on the app 1`] = `
<div
  class="App"
  data-testid="test-all"
>
  <div
    class="header"
  >
    <div
      class="popup "
    >
      <div
        class="quantity-info"
      >
        <div>
          <h1>
            My Shopping Cart
          </h1>
          <div
            class="scroll"
          />
          <div>
            <h3>
              Total: $ 
              0
            </h3>
            <button
              class="checkout-btn"
            >
              Checkout
            </button>
            <button
              class="bck-btn"
            >
              Back
            </button>
          </div>
        </div>
        <div>
          <button>
            X
          </button>
        </div>
      </div>
    </div>
    <p
      class="logo-style"
      style="color: rgb(255, 203, 164);"
    >
      Dura_glam Store
    </p>
    <div
      class="left-header"
    >
      <ul>
        <a
          href="/shopping-cart/"
        >
          <li>
            Home
          </li>
        </a>
        <a
          class="link"
          data-testid="test-app"
          href="/shopping-cart/store"
        >
          <li>
            Store
          </li>
        </a>
        <a
          class=""
          href="/shopping-cart/about"
        >
          <li>
            About
          </li>
        </a>
      </ul>
      <div
        class="svg-container"
      >
        <div
          class="svg-background"
        >
          <img
            alt="cart icon"
            class="cart"
            src="cart-variant.svg"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    style="background-color: rgb(0, 0, 0); padding: 10px; position: absolute; bottom: 0px; right: 0px; left: 0px; color: white; display: flex; justify-content: center;"
  >
    Copyright Â© 2023 Olanrewaju
  </div>
</div>
`;
